payment_ippay/models/payment.py

```diff
@@ -30,29 +38,30 @@ def ippay_s2s_form_process(self, data):
                 "partner_id": int(data.get("partner_id")),
             }
             # Only create a new Token if it doesn't already exist
-            token_code = Token._ippay_get_token(values)
-            PaymentMethod = Token.sudo().search(
+            token_code = token._ippay_get_token(values)
+            payment_method = token.sudo().search(
                 [
                     ("acquirer_ref", "=", token_code),
                     ("partner_id", "=", values.get("partner_id")),
                     ("acquirer_id", "=", values.get("acquirer_id")),
                 ],
                 limit=1,
             )
-            if not PaymentMethod:
-                PaymentMethod = Token.sudo().create(values, token_code)
+            if not payment_method:
+                payment_method = token.sudo().create(values)
         else:
-            PaymentMethod = Token.sudo().search(
+            payment_method = token.sudo().search(
                 [
                     ("id", "=", int(token_id)),
-                    ("partner_id", "=", values.get("partner_id")),
-                    ("acquirer_id", "=", values.get("acquirer_id")),
+                    ("partner_id", "=", data.get("partner_id")),
+                    ("acquirer_id", "=", data.get("acquirer_id")),
```

Comment:
Looks like we had a bug here.

Reply:
values is inside 'if' condition and search call in else part